frontend/src/pages/modelRegistrySettings/ProjectsTab/ProjectsSettingsTab.tsx

```diff
@@ -0,0 +1,179 @@
+  roleBindingPermissionsRB: ContextResourceData<RoleBindingKind>;
+  permissionOptions: {
+    type: RoleBindingPermissionsRoleType;
+    description: string;
+  }[];
+  projectName: string;
+  roleRefName?: RoleBindingRoleRef['name'];
+  labels?: { [key: string]: string };
+  isProject?: boolean;
+};
+
+const ProjectsSettingsTab: React.FC<RoleBindingProjectPermissionsProps> = ({
+  ownerReference,
+  roleBindingPermissionsRB,
+  permissionOptions,
+  projectName,
+  roleRefName,
+  labels,
+  isProject,
+}) => {
+  const [addField, setAddField] = React.useState(false);
+  const [error, setError] = React.useState<Error | undefined>(undefined);
+  const {
+    data: roleBindings,
+    loaded,
+    error: loadError,
+    refresh: refreshRB,
+  } = roleBindingPermissionsRB;
+
+  const { projects } = React.useContext(ProjectsContext);
+  const filteredProjects = projects.filter(
+    (project) => !removePrefix(roleBindings).includes(project.metadata.name),
+  );
+
+  if (loadError) {
+    return (
+      <EmptyState
+        variant={EmptyStateVariant.lg}
+        data-id="error-empty-state"
+        id={ProjectSectionID.PERMISSIONS}
+      >
+        <EmptyStateHeader
+          titleText="There was an issue loading projects"
+          icon={<EmptyStateIcon icon={ExclamationCircleIcon} />}
+          headingLevel="h2"
+        />
+        <EmptyStateBody>{loadError.message}</EmptyStateBody>
+      </EmptyState>
+    );
+  }
+
+  if (!loaded) {
+    return (
+      <EmptyState
+        variant={EmptyStateVariant.lg}
+        data-id="loading-empty-state"
+        id={ProjectSectionID.PERMISSIONS}
+      >
+        <Spinner size="xl" />
+        <EmptyStateHeader titleText="Loading" headingLevel="h2" />
+      </EmptyState>
+    );
+  }
+
+  return (
+    <PageSection isFilled variant="light">
+      <Stack hasGutter>
+        <StackItem>
+          <Flex
+            direction={{ default: 'row' }}
+            gap={{ default: 'gapSm' }}
+            alignItems={{ default: 'alignItemsCenter' }}
+          >
+            <HeaderIcon type={ProjectObjectType.project} />
+            <FlexItem>
+              <Title headingLevel="h2" size="xl">
+                Projects
+              </Title>
+            </FlexItem>
+          </Flex>
+        </StackItem>
+        <StackItem>
+          <RoleBindingPermissionsTable
+            ownerReference={ownerReference}
+            permissions={filterRoleBindingSubjects(
+              roleBindings,
+              RoleBindingPermissionsRBType.GROUP,
+              isProject,
+            )}
+            permissionOptions={permissionOptions}
+            namespace={projectName}
+            roleRefKind="Role"
+            isProject={isProject}
+            roleRefName={roleRefName}
+            labels={labels}
+            typeAhead={
+              filteredProjects.length > 0
+                ? filteredProjects.map((project) => project.metadata.name)
+                : undefined
+            }
```

Comment:
So as you have this implemented, the options in the dropdown are project names (raw namespaces) rather than project display names.

It's going to be tricky... but we should show the display name of projects in the dropdown and the non-edit-mode cell of the table instead of the `project.metadata.name`.

One way to do that might be to add an optional `getSubjectDisplayName` callback prop that takes a string (the `project.metadata.name`) and returns the project's display name. You'd define it here since you already have projects in memory and you could do the lookup here (by finding the matching project object and passing it to `getDisplayNameFromK8sResource` imported from `~/concepts/k8s/utils`). Then you'd feed that `getSubjectDisplayName` prop all the way down to where your select options are rendered in `RoleBindingPermissionsNameInput` and use it to render the display name as children of `<SelectOption>`, and also use it in the non-editing mode of `RoleBindingPermissionsTableRow` instead of directly displaying `roleBindingName`. If it's not passed in it could just default to `roleBindingName`.

Reply:
oh sorry. I didn't notice that. I will update this.

Reply:
No worries! Thanks for the hard work here.